orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -39,12 +44,16 @@ def execution_session_run_forward(execution_session, onnx_model, gradient_accumu
         forward_inputs = C.OrtValueVector()
         forward_inputs.reserve(len(inputs))
         for input in inputs:
-            forward_inputs.push_back(to_dlpack(input), input.dtype == torch.bool)
+            forward_inputs.push_back(torch_tensor_to_dlpack(input), input.dtype == torch.bool)
 
         forward_outputs = C.OrtValueVector()
         # Run and return module outputs.
         execution_session.run_forward(forward_inputs, forward_outputs, state, gradient_accumulation_manager.cache)
-        user_outputs = gradient_accumulation_manager.extract_outputs_and_maybe_update_cache(forward_outputs)
+        device = torch.cpu() if len(inputs) == 0 else inputs[0].device
```

Comment:
Consider using `https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/ortmodule/_utils.py#L98` instead as opposed to hard code